### PR TITLE
:recycle: Improve config implementation

### PR DESCRIPTION
**Pull request by rubenaeg**
## Proposed changes
This PR refactors the config implementations, by merging type definitions with actual implementation, thus keeping a clean implementation for both `ProjectConfig` and `UserConfig`. This will also allow for some validation at a later stage.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: core/src/ProjectConfig.ts [hunk 1]
```diff
@@ -1,13 +1,120 @@
...
+  /**
+   * Returns configuration, considering the stage. If no stage is set, just returns this.getConfigContent().
+   */
+  private load(projectPath: string, stage?: string): ProjectConfig {
+    const configContent: ProjectConfig = this.loadContent(projectPath);
+
+    if (stage) {
+      const stagedConfig: ProjectConfig | undefined = _get(configContent, `stages.${stage}`);
+
+      if (stagedConfig) {
```
Comment (aswetlow) [inline core/src/ProjectConfig.ts]: Print warning if the user expects a stage config, but there is no stage config. Not sure about this one.

File: core/src/UserConfig.ts [hunk 2]
```diff
@@ -4,47 +4,64 @@ import _get from 'lodash.get';
...
 export class UserConfig {
-  private config: JovoUserConfigFile;
+  webhook!: {
+    uuid: string;
+  };
+  cli!: {
+    plugins: string[];
+    presets: Preset[];
+  };
+  timeLastUpdateMessage?: string | number;
```
Comment (aswetlow) [inline core/src/UserConfig.ts]: Why is it `string `or `number`?

Comment (rubenaeg) [reply core/src/UserConfig.ts]: Legacy code, will be removed :+1:

File: core/src/ProjectConfig.ts (hunk 1 above)
Comment (rubenaeg) [reply core/src/ProjectConfig.ts]: Good idea, will be implemented :+1:

**Review by jankoenig — APPROVED** [review]

**Pull request merged**